src/Compilers/VisualBasic/Portable/Compilation/SyntaxTreeSemanticModel.vb

```diff
@@ -864,7 +864,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' </summary>
         Private Function CheckSymbolLocationsAgainstSyntax(symbol As NamedTypeSymbol, nodeToCheck As VisualBasicSyntaxNode) As NamedTypeSymbol
             For Each location In symbol.Locations
-                If location.SourceTree Is Me.SyntaxTree AndAlso nodeToCheck.Span.Contains(location.SourceSpan.Start) Then
+                If location.SourceTree Is Me.SyntaxTree AndAlso nodeToCheck.Span.IntersectsWith(location.SourceSpan.Start) Then
                     Return symbol
```

Comment:
`location` is coming from the type name token. This doesn't work when the user is in the process of declaring the type because you get something like this:
```
Partial Class $$
```
The location of the name is where the cursor is placed, which also happens to be the exact end of the span. `Contains` treats the span as a right-open interval, so checks fails. `IntersectsWith` behaves as if the span is a closed interval, and the check succeeds.